packages/core-components/src/layout/Sidebar/Bar.tsx

```diff
@@ -150,71 +132,57 @@ export function Sidebar(props: PropsWithChildren<Props>) {
 
   const setOpen = (open: boolean) => {
     if (open) {
-      handleOpen();
+      setState(State.Open);
+      toggleSidebarPinState();
     } else {
-      handleClose();
+      setState(State.Closed);
+      toggleSidebarPinState();
     }
   };
 
   return (
-    <div
-      className={classes.root}
-      data-testid="sidebar-root"
-      onMouseEnter={disableExpandOnHover ? () => {} : handleOpen}
-      onFocus={disableExpandOnHover ? () => {} : handleOpen}
-      onMouseLeave={disableExpandOnHover ? () => {} : handleClose}
-      onBlur={disableExpandOnHover ? () => {} : handleClose}
+    <SidebarContext.Provider
+      value={{
+        isOpen,
+        setOpen,
+      }}
     >
-      <SidebarContext.Provider
-        value={{
-          isOpen,
-          setOpen,
-        }}
+      <div
+        onMouseEnter={handleOpen}
+        onFocus={handleOpen}
+        onMouseLeave={handleClose}
+        onBlur={handleClose}
+        data-testid="sidebar-root"
+        className={classnames(classes.drawer, {
+          [classes.drawerOpen]: isOpen,
+        })}
       >
-        <div
-          className={classnames(classes.drawer, {
-            [classes.drawerOpen]: isOpen,
-          })}
-        >
-          {children}
-        </div>
-      </SidebarContext.Provider>
-    </div>
+        {children}
+      </div>
+    </SidebarContext.Provider>
   );
-}
+};
 
-/**
- * A button which allows you to expand the sidebar when clicked.
- * Use optionally to replace sidebar's expand-on-hover feature with expand-on-click.
- *
- * @public
- */
-export const SidebarExpandButton = () => {
-  const classes = useStyles();
-  const { isOpen, setOpen } = useContext(SidebarContext);
-  const { isPinned } = useContext(SidebarPinStateContext);
-  const isSmallScreen = useMediaQuery<BackstageTheme>(theme =>
-    theme.breakpoints.down('md'),
+export const Sidebar = ({
+  children,
+  openDelayMs,
+  closeDelayMs,
+  disableExpandOnHover,
+}: React.PropsWithChildren<Props>) => {
+  const isMobileScreen = useMediaQuery<BackstageTheme>(
+    theme => theme.breakpoints.down('xs'),
+    { noSsr: true },
```

Comment:
only these need the nossr parameter, not elsewhere?

Reply:
Might have to be investigated further, but I think that in general all `useMediaQuery` & `withWidth` should have a `noSsr` flag to improve performance: https://v4.mui.com/components/use-media-query/#usemediaquery-query-options-matches. 

Only changed those in the scope of the PR here :)